tests/controller/README.md

```diff
@@ -1 +1,209 @@
-This folder contains steps to deploy the controller framework with an example manifest.yaml
\ No newline at end of file
+##What is a controller?
+
+In simple terms, controller is a framework to keep the registry up-to-date. The idea for this framework is inspired from the [Kubernetes controller](https://kubernetes.io/docs/concepts/architecture/controller/) pattern. Controller can be used to enrich the API data stored in the registry, with additional analytical artifacts. 
+
+### How does it work?
+* A Controller will take in configuration (aka manifest) which describes the desired state of the registry. The desired state in this context is essentially dependency relations between different entities in the database. 
+* The controller’s job is to keep these dependency relations valid and up-to-date. It will take actions to eventually get the registry to it's desired state.
+* Example manifest:
+  ```
+  id: "example-manifest"
+  generated_resources:
+    - pattern: apis/-/versions/-/specs/-/artifacts/lint-spectral
+      dependencies:
+        - pattern: $resource.spec
+          filter: "mime_type.contains('openapi')"
+      action: "registry compute lint $resource.spec --linter spectral"
+  ```
+  The above manifest defines that for every spec (`pattern: $resource.spec`) of type openapi (`filter: "mime_type.contains('openapi')"`) in the registry, there should exist a corresponding lint-spectral artifact (`pattern: apis/-/versions/-/specs/-/artifacts/lint-spectral`). 
+  If the target artifact is not present or out-of-date, then it can be re-generated using the specified action (`action: "registry compute lint $resource.spec --linter spectral"`).
+
+* The controller will compare the current and the supplied desired state of the registry. For resources which are non-existing or outdated, the controller will execute the action supplied in the manifest.
+
+### How to use it?
+
+With this basic definition in mind, you can supply different configurations to the controller to generate various artifacts in the registry.
+
+#### Setup the registry server:
```

Comment:
Consider removing the `:` from the headers. The formatting of headers makes it unnecessary and it doesn't seem to be consistently included.

Reply:
Done.